src/services/sanity.js

```diff
@@ -1312,6 +1312,63 @@ export async function fetchRelatedLessons(railContentId, brand) {
+  const parentProjections = `railcontent_id, _type, parent_type, parent_content_data, difficulty`
+  const parentQuery = `*[railcontent_id == ${railContentId} && brand == "${brand}" && (!defined(permission) || references(*[_type=='permission']._id))]
+    {${parentProjections}}`
+  const parentQueryResult = await fetchSanity(parentQuery, false)
+
+  //get fields from parent query
+  const parentType = parentQueryResult.parent_type
+  const parentId = parentQueryResult.parent_content_data[0].id
+  const difficulty = parentQueryResult.difficulty
+
+  //genreQuery. all this to get a stringified array of genres
+  const genreQuery = `*[_type == "${parentType}" && brand == "${brand}" && railcontent_id == ${parentId} && _id != "drafts."+"${parentType}_${parentId}"][0]{"genre":genre[]->_id}`
+  const genreResult = await fetchSanity(genreQuery, true)
+  // const genreArray = Object.values(genreResult).map(obj => obj.name).join(', ') //this is giving me pain. I need to convert object -> string
+  const genreString = JSON.stringify(genreResult['genre']).replace(/[\[\]]/g, '')
+
+  //get the related_lessons field as an object
+  const tutorialFilter = await getSubQueryFilter(parentType, brand, parentId, difficulty, genreString)
+  const quickTipFilter = await getSubQueryFilter('quick-tips', brand, parentId, difficulty)
+  const songFilter = await getSubQueryFilter('song', brand, parentId, difficulty, genreString)
+  const relatedLessons = await fetchSanity(`array::unique([
+      ...*[${tutorialFilter}]{${defaultProjections}}|order(published_on desc, title asc)[0...10],
+      ...*[${quickTipFilter}]{${defaultProjections}}|order(published_on desc, title asc)[0...10],
+      ...*[${songFilter}]{${defaultProjections}}|order(published_on desc, title asc)[0...10],
+    ])`, true)
+
+  //merge into one json object
+  return {...parentQueryResult,
+    related_lessons: relatedLessons
+  }
+
+/*  async function fetchSubQueryByType(type, brand, id, genres, difficulty) {
+    const subFilter = await new FilterBuilder(`_type == "${type}" && brand == "${brand}" && railcontent_id != ${id} && references([${genres}]) && difficulty == ${difficulty}`).buildFilter()
+    const subQuery = `*[${subFilter}]{${defaultProjections}}|order(published_on desc, title asc)[0...10]`
+    return fetchSanity(subQuery, true)
+  }*/
+
+  async function getSubQueryFilter(type, brand, id, difficulty, genres = null){
+    if (genres) {
+      return new FilterBuilder(`_type == "${type}" && brand == "${brand}" && railcontent_id != ${id} && references([${genres}]) && difficulty == ${difficulty}`).buildFilter()
+    } else {
+      return new FilterBuilder(`_type == "${type}" && brand == "${brand}" && railcontent_id != ${id} && difficulty == ${difficulty}`).buildFilter()
+    }
+  }
```

Comment:
My preference for extending this type of query, and programming in general. A good rule of thumb is that if you're writing the same code twice, you can parameterize (2) or make a new method (3+ times).
```suggestion
    const genresString = genres ? ` && references([${genres}])` : ''
    return new FilterBuilder(`_type == "${type}" && brand == "${brand}" && railcontent_id != ${id} && difficulty == ${difficulty} ${genreString}`).buildFilter()
    }
  }
```